java/vector/src/test/java/org/apache/arrow/vector/ipc/BaseFileTest.java

```diff
@@ -846,4 +862,293 @@ protected void validateListAsMapData(VectorSchemaRoot root) {
+        vectorB.setSafe(1, "lorem".getBytes(StandardCharsets.UTF_8));
+        vectorC.setSafe(0, "qui".getBytes(StandardCharsets.UTF_8));
+        vectorC.setSafe(1, "dolor".getBytes(StandardCharsets.UTF_8));
+        vectorD.setSafe(0, "amet".getBytes(StandardCharsets.UTF_8));
+        if (state == 7) {
+          vectorD.setSafe(1, "quia".getBytes(StandardCharsets.UTF_8));
+        }
+
+        root.setRowCount(2);
+        arrowWriter.writeBatch();
+
+        // batch 3
+        vectorA.setNull(0);
+        vectorA.setNull(1);
+        vectorB.setSafe(0, "ipsum".getBytes(StandardCharsets.UTF_8));
+        vectorB.setNull(1);
+        vectorC.setNull(0);
+        vectorC.setSafe(1, "qui".getBytes(StandardCharsets.UTF_8));
+        vectorD.setNull(0);
+        if (state == 7) {
+          vectorD.setSafe(1, "quia".getBytes(StandardCharsets.UTF_8));
+        }
+
+        root.setRowCount(2);
+        arrowWriter.writeBatch();
+
+        // batch 4
+        vectorA.setSafe(0, "bar".getBytes(StandardCharsets.UTF_8));
+        vectorA.setSafe(1, "zap".getBytes(StandardCharsets.UTF_8));
+        vectorB.setNull(0);
+        vectorB.setSafe(1, "lorem".getBytes(StandardCharsets.UTF_8));
+        vectorC.setNull(0);
+        vectorC.setNull(1);
+        if (state == 7) {
+          vectorD.setSafe(0, "quia".getBytes(StandardCharsets.UTF_8));
+        }
+        vectorD.setNull(1);
+
+        root.setRowCount(2);
+        arrowWriter.writeBatch();
+
+        arrowWriter.end();
+      } catch (Exception e) {
+        if (arrowWriter != null) {
+          arrowWriter.close();
+        }
+        throw e;
+      }
+    }
+  }
+
+  Map<Integer, String[][]> valuesPerBlock = new HashMap<Integer, String[][]>();
+
+  {
+    valuesPerBlock.put(0, new String[][]{
+        {"foo", "bar"},
+        {"lorem", "ipsum"},
+        {null, null},
+        {"porro", "amet"}
+    });
+    valuesPerBlock.put(1, new String[][]{
+        {"meep", "bar"},
+        {"ipsum", "lorem"},
+        {"qui", "dolor"},
+        {"amet", "quia"}
+    });
+    valuesPerBlock.put(2, new String[][]{
+        {null, null},
+        {"ipsum", null},
+        {null, "qui"},
+        {null, "quia"}
+    });
+    valuesPerBlock.put(3, new String[][]{
+        {"bar", "zap"},
+        {null, "lorem"},
+        {null, null},
+        {"quia", null}
+    });
+  }
+
+  protected void assertDictionary(FieldVector encoded, ArrowReader reader, String... expected) throws Exception {
+    DictionaryEncoding dictionaryEncoding = encoded.getField().getDictionary();
+    BaseDictionary dictionary = reader.getDictionaryVectors().get(dictionaryEncoding.getId());
+    try (ValueVector decoded = DictionaryEncoder.decode(encoded, dictionary)) {
+      Assertions.assertEquals(expected.length, encoded.getValueCount());
+      for (int i = 0; i < expected.length; i++) {
+        if (expected[i] == null) {
+          Assertions.assertNull(decoded.getObject(i));
```

Comment:
It would be good to be consistent about static imports of assertion functions. assertNotNull() is statically imported but assertNull() isn't.

Reply:
ack